### ISSUE TITLE
why not get logits from memory[:,-1, :] instead of func(memory[:,:mem_slots,:])

### ISSUE THREAD

**Issue by zhaoguangxiang** [L0SG/relational-rnn-pytorch#4]
It is also reasonable

**Comment by L0SG** [comment]
Assuming that you mean extracting logits from the memory by `next_memory = next_memory[:, :-n, :]` then `output = next_memory.view(next_memory.shape[0], -1)` [in here](https://github.com/L0SG/relational-rnn-pytorch/blob/1b16ae32988625b16b95f920b0f6fe55ed4e45e7/relational_rnn_models.py#L333), it is a standard practice to do so (like in the "normal" RNN literature, where we get outputs (logits) from RNN's hidden states, and not from the input vector directly).

That said, `output = memory[:,-1, :]` would mean that we just treat the (self-attended) input vector directly as the output logit which is not consistent with the paper I believe. Please also refer to the [original Sonnet implementation](https://github.com/deepmind/sonnet/blob/56c917e156d84db2bcbc1f027ccbeae3cb1192cf/sonnet/python/modules/relational_memory.py#L263) which does the same thing (`next_memory = next_memory[:, :-n, :]` then ` output = basic.BatchFlatten()(next_memory)`).
